packages/frontend/editor-ui/src/components/ImportWorkflowUrlModal.vue

```diff
@@ -0,0 +1,83 @@
+<script setup lang="ts">
+import { ref, watch } from 'vue';
+import { useI18n } from '@/composables/useI18n';
+import { useUIStore } from '@/stores/ui.store';
+import { nodeViewEventBus } from '@/event-bus';
+import { VALID_WORKFLOW_IMPORT_URL_REGEX, IMPORT_WORKFLOW_URL_MODAL_KEY } from '@/constants';
+
+const i18n = useI18n();
+const uiStore = useUIStore();
+
+const url = ref('');
+const isValid = ref(true);
+
+const closeModal = () => {
+	uiStore.closeModal(IMPORT_WORKFLOW_URL_MODAL_KEY);
+};
+
+const confirm = () => {
+	if (!url.value.match(VALID_WORKFLOW_IMPORT_URL_REGEX)) {
+		isValid.value = false;
+		return;
+	}
+	nodeViewEventBus.emit('importWorkflowUrl', { url: url.value });
+	uiStore.closeModal(IMPORT_WORKFLOW_URL_MODAL_KEY);
+};
+
+watch(url, (newValue) => {
+	if (!newValue) {
+		isValid.value = true;
+	} else {
+		isValid.value = VALID_WORKFLOW_IMPORT_URL_REGEX.test(newValue);
+	}
+});
+</script>
+
+<template>
+	<Modal
+		:name="IMPORT_WORKFLOW_URL_MODAL_KEY"
+		:title="i18n.baseText('mainSidebar.prompt.importWorkflowFromUrl')"
+		:show-close="true"
+		:center="true"
+		width="420px"
+	>
+		<template #content>
+			<div>
+				<n8n-input
+					v-model="url"
+					:placeholder="i18n.baseText('mainSidebar.prompt.workflowUrl')"
+					:state="isValid ? 'default' : 'error'"
+					@input="isValid = true"
+				/>
+				<p v-if="!isValid" :class="$style.error - text">
+					{{ i18n.baseText('mainSidebar.prompt.invalidUrl') }}
+				</p>
+			</div>
+		</template>
+		<template #footer>
+			<div :class="$style.footer">
+				<n8n-button type="secondary" float="right" @click="closeModal">
+					{{ i18n.baseText('mainSidebar.prompt.cancel') }}
+				</n8n-button>
+				<n8n-button type="primary" float="right" @click="confirm">
```

Comment:
It would be nice to disable the button when url is not valid:
```suggestion
				<n8n-button type="primary" float="right" :disabled="!url.value || !isValid.value" @click="confirm">
```